### PR TITLE
[5.8] Add redirectAction to Controller

### PR DESCRIPTION
**Pull request by barryvdh**
This adds a method to simplify returning a redirect to a method in the current class. So basically `return $this->redirectAction('index')` would be equivalent to `return redirect()->action([static::class, 'index']);`

Not sure if the name is best though (could be just `action` or `redirectToAction`)

### CODE REVIEW

**Comment by driesvints** [comment]
Why not just talk to the method in the controller directly?

```php
return $this->index();
```

File: src/Illuminate/Routing/Controller.php [hunk 1]
```diff
@@ -54,6 +54,20 @@ public function callAction($method, $parameters)
...
     }
 
+    /**
+     * Return a redirect to a method in this Controller.
+     *
+     * @param $method
+     * @param array $parameters
+     * @param int $status
+     * @param array $headers
+     * @return \Illuminate\Http\RedirectResponse
```
Comment (driesvints) [inline src/Illuminate/Routing/Controller.php]: Can you format these params like we do in the rest of the framework? Also add a type-hint for the `$method` param.

**Review by driesvints — CHANGES_REQUESTED** [review]

**Comment by barryvdh** [comment]
It's a redirect, eg back to index after updating or to the edit page etc.

**Comment by barryvdh** [comment]
Although perhaps it could just return an url, so you can use it more broadly. Eg. `$this->action ('index')`, so it's easier to pass to views, json responses etc.

**Comment by driesvints** [comment]
Yeah but you can just use the `action` helper for that already. 

```php
return redirect(action([static::class, 'index']));
```

Honestly I'm really a fan of these "small" helper methods. There's about a dozen other use cases for redirect & route helpers which could go into the controller but are just a bit too much to maintain.

**Comment by barryvdh** [comment]
Yeah thats what I said in the opening post, but IMHO that's not the most 'readable' form, as referencing actions in the current controller is a very common action (at least in my projects). But I understand it's something that could be relatively easily done in your own project..

**Comment by driesvints** [comment]
@barryvdh yeah no worries. Appreciate you suggesting it :)

**Pull request closed without merging**